core/src/main/java/org/apache/calcite/avatica/Meta.java

```diff
@@ -577,6 +577,12 @@ public static MetaResultSet create(String connectionId, int statementId,
           Objects.requireNonNull(signature), firstFrame, -1L);
     }
 
+    public static MetaResultSet create(String connectionId, int statementId,
+        boolean ownStatement, Signature signature, Frame firstFrame, long updateCount) {
+      return new MetaResultSet(connectionId, statementId, ownStatement,
+          Objects.requireNonNull(signature), firstFrame, updateCount);
+    }
+
```

Comment:
Hmm, this is OK. But why can't we make `MetaResultSet` constructor public? 
@julianhyde Do you know the reason?

Reply:
If things are public it is easier for the user, harder for the maintainer. So my inclination is always to make things private, unless there is a good reason.

This is what I call a "drill a hole" change. Making something public without any rationale and without tests.